eclair-core/src/main/scala/fr/acinq/eclair/PluginParams.scala

```diff
@@ -55,4 +58,15 @@ trait CustomCommitmentsPlugin extends PluginParams {
    * returned by this method.
    */
   def getHtlcsRelayedOut(htlcsIn: Seq[IncomingHtlc], nodeParams: NodeParams, log: LoggingAdapter): Map[Origin, Set[(ByteVector32, Long)]]
+}
+
+// @formatter:off
+case class InterceptOpenChannelReceived(replyTo: ActorRef[InterceptOpenChannelResponse], open: Either[OpenChannel, OpenDualFundedChannel], temporaryChannelId: ByteVector32, localParams: LocalParams, fundingAmount: Satoshi)
+sealed trait InterceptOpenChannelResponse
+case class AcceptOpenChannel(temporaryChannelId: ByteVector32, localParams: LocalParams) extends InterceptOpenChannelResponse
```

Comment:
One of the important decisions that the plugin makes will be how much to contribute (when using dual funding), so we should probably add the necessary field right now to avoid breaking plugins later.

Also, more generally, why do we need to include `LocalParams` in `InterceptOpenChannelReceived` and `AcceptOpenChannel`? When looking at the fields inside this class, I don't think any of them are useful for the plugin, and I don't think the plugin should modify them, they directly come from the values the node operators chose in `eclair.conf` (and changing some of them would actually break the channel logic...)? I think I was the one to initially suggest passing `LocalParams`, but now that I look more into it I'm not sure it makes sense, unless you have a use-case for it?

```suggestion
case class AcceptOpenChannel(temporaryChannelId: ByteVector32, localFundingAmount: Satoshi) extends InterceptOpenChannelResponse
```

Reply:
I took a look at the channel interceptor support in CLN and LND to create a table of possible response fields for the Eclair channel interceptor: [channel_interceptor.md](https://gist.github.com/remyers/df02cb7eb938b29dc8715b7fa3098bac)

Perhaps you we can brainstorm around this topic a bit to make sure we have a comprehensive solution.

Reply:
Good idea, I've answered in my top-level comment, there are some fields that we can indeed make configurable!

In the longer term, it may be interesting to also let those plugins do the channel funding part to let plugins implement things like vortex / coinjoin, but that will require more changes to the `Channel` logic so we'll wait to finalize dual funding / liquidity ads before adding that capability.